React/wbooks/src/BookDetail/index.js

```diff
@@ -1,37 +1,55 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { Component } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
 
 import { BOOKS } from '../books_json';
+import { bookDetailAction } from '../redux/bookDetail/actions';
 
 import BookDetailInfo from './Components/BookDetailInfo';
 import BookSuggestions from './Components/BookSuggestions';
 import CommentSection from './Components/CommentSection';
 
-function BookDetail({ match }) {
-  const bookId = match.params.id;
-  const book = BOOKS.find(b => b.id == bookId);
-  const suggestions = BOOKS.filter(b => b.genre == book.genre);
+class BookDetail extends Component {
+  componentWillMount() {
```

Comment:
componentWillMount will be deprecated in the next React version. Avoid using it. Requests should be done in componentDidMount